validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/ValidatorDutyMetrics.java

```diff
@@ -38,9 +39,22 @@ public static ValidatorDutyMetrics create(final MetricsSystem metricsSystem) {
   }
 
   public SafeFuture<DutyResult> performDutyWithMetrics(final Duty duty) {
-    final String dutyType = duty.getType().getType();
-    final OperationTimer timer = dutyMetric.labels(dutyType, "total");
-    final OperationTimer.TimingContext context = timer.startTimer();
+    final OperationTimer.TimingContext context = startTimer(getDutyType(duty), "total");
     return duty.performDuty().alwaysRun(context::stopTimer);
   }
+
+  private OperationTimer.TimingContext startTimer(final String dutyType, final String step) {
+    final OperationTimer timer = dutyMetric.labels(dutyType, step);
+    return timer.startTimer();
+  }
+
+  private static String getDutyType(final Duty duty) {
+    return duty.getType().getType();
+  }
+
+  public <T> SafeFuture<T> record(
+      final Supplier<SafeFuture<T>> supplier, final String dutyType, final String step) {
```

Comment:
I think we could change `dutyType` from `String` to `Duty` and then use the same `getDutyType(duty)` method to get the type.

This should make this a bit more consistent and there will be no need to call `final String dutyType = getType().getType();` outside of the scope of this method.

I think I'd also rename `supplier` to something a bit more meaningful like `dutyStepFutureSupplier`.